src/util/prolog/prologResult.ts

```diff
@@ -0,0 +1,120 @@
+import * as ld from "lodash";
+
+// First Group will be the variable value. Doesn't fetch the variable names
+const prologAnswerPattern: string = " = (\\w*|\\[.*\\])?(,| ;|.)";
+
+export class PrologResult {
+	private readonly rawResults: string[];
+	private readonly results: Map<string, Map<number, string|boolean>>;
+
+	constructor(rawResults: string[]) {
+		this.rawResults = rawResults;
+		this.results = new Map();
+	}
+
+	public getResults(): Map<string, Map<number, string|boolean>> {
+		if (this.results.size !== 0) { return this.results; }
```

Comment:
But what if `results` is supposed to be empty? I think it'd be nicer to change the type of `results` to `Map<...> | undefined`, and check if it is `undefined` here.